src/main/kotlin/com/exactpro/th2/conn/dirty/tcp/core/api/impl/Channel.kt

```diff
@@ -168,11 +170,13 @@ class Channel(
             // Date from buffer should be copied for prost-processing (mangler.postOutgoing and onMessage handling).
             // The post-processing is executed asynchronously after sending message via tcp channel where original buffer is released
             val data = Unpooled.copiedBuffer(buffer).asReadOnly()
+            val sendingTimestamp = Instant.now()
             thenRunAsync({
                 runCatching {
                     logger.trace { "Post process of '${messageId.toJson()}' message id: ${hexDump(data)}" }
                     if (mode.mangle) mangler.postOutgoing(this@Channel, data, metadata)
-                    val resolvedMessageId = runCatching { onMessage.accept(data, messageId, metadata, eventId) }
+                    val finalMetadata = metadata.add(IChannel.OPERATION_TIMESTAMP_PROPERTY, sendingTimestamp.toString())
```

Comment:
metadata has MutableMap type, could we use put method instead of extension method add